### PR TITLE
client: Jwt token based auth for engine api enabled via `rpcEngineAuth` arg

### PR DESCRIPTION
**Pull request by g11tech**
Kiln spec v2 has jwt based auth for the engine json rpc requests through both http/websocket protocols. Following are some salient features of implementing same in this PR:
- jwt token auth for http and websocket upgrade requests
- iat (issued at) +- 5 seconds from current time
- `--jwt-secret` param to specify the file for jwt secret, in its absence a random 256 bit key is generated (and dumped in a file)
- `--rpcEngineAuth`  param addition to enable the above authentication mechanism, so that the requests remain backward compatible till all clients move to jwt auth

This PR is a fresh pull in continuation to this previous PR https://github.com/ethereumjs/ethereumjs-monorepo/pull/1723
cc: @ryanio

### CODE REVIEW

**Comment by ryanio** [comment]
looks great, thanks so much for doing and sticking through! can't believe you also got the ws auth working, that is very cool. we're just missing some coverage on the ws auth now, would you be able to add a few cases to the rpc.spec tests to get coverage on the lines? then I'll approve and merge in :)

File: packages/client/test/rpc/websocket.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+import tape from 'tape'
+import { encode, TAlgorithm } from 'jwt-simple'
+import { startRPC, closeRPC } from './helpers'
+import { METHOD_NOT_FOUND } from '../../lib/rpc/error-code'
+const request = require('superwstest')
+
+const jwtSecret = Buffer.from(Array.from({ length: 32 }, () => Math.round(Math.random() * 255)))
+const wsPort = 3000
+
+tape('call JSON-RPC auth protected server with valid token ', (t) => {
```
Comment (ryanio) [inline packages/client/test/rpc/websocket.spec.ts:10]:
```suggestion
tape('call JSON-RPC auth protected server with valid token', (t) => {
```

File: packages/client/bin/cli.ts [hunk 2]
```diff
@@ -122,6 +120,14 @@ const args = yargs(hideBin(process.argv))
     string: true,
     default: 'localhost',
   })
+  .option('rpcEngineAuth', {
+    describe: 'Enable jwt authentication for Engine RPC server',
+    boolean: false,
```
Comment (ryanio) [inline packages/client/bin/cli.ts:125]:
```suggestion
    boolean: true,
    default: false
```

**Review by ryanio — APPROVED** [review]
looks fantastic, thank you! will merge in after ci completes

**Comment by g11tech** [comment]
> looks fantastic, thank you! will merge in after ci completes

thank you @ryanio :heart: happy to contribute to ethereumjs! :rocket:

**Pull request merged**